source/common/http/http3/conn_pool.cc

```diff
@@ -17,30 +17,62 @@ namespace Envoy {
 namespace Http {
 namespace Http3 {
 
+// Http3 subclass of FixedHttpConnPoolImpl which exists to store quic data.
+class Http3ConnPoolImpl : public FixedHttpConnPoolImpl {
+public:
+  Http3ConnPoolImpl(Upstream::HostConstSharedPtr host, Upstream::ResourcePriority priority,
+                    Event::Dispatcher& dispatcher,
+                    const Network::ConnectionSocket::OptionsSharedPtr& options,
+                    const Network::TransportSocketOptionsSharedPtr& transport_socket_options,
+                    Random::RandomGenerator& random_generator,
+                    Upstream::ClusterConnectivityState& state, CreateClientFn client_fn,
+                    CreateCodecFn codec_fn, std::vector<Http::Protocol> protocol,
+                    TimeSource& time_source)
+      : FixedHttpConnPoolImpl(host, priority, dispatcher, options, transport_socket_options,
+                              random_generator, state, client_fn, codec_fn, protocol) {
+    auto source_address = host_->cluster().sourceAddress();
+    if (!source_address.get()) {
+      auto host_address = host->address();
+      source_address = Network::Utility::getLocalAddress(host_address->ip()->version());
+    }
+    Network::TransportSocketFactory& transport_socket_factory = host->transportSocketFactory();
+    quic_info_ =
+        Config::Utility::getAndCheckFactoryByName<Http::QuicClientConnectionFactory>(
+            Http::QuicCodecNames::get().Quiche)
```

Comment:
example of cross dependency issue that we were discussing in the slack thread (so yeah, I think I'm firmly in the no extension camp now!)